ddtrace/internal/_encoding.pyx

```diff
@@ -569,8 +569,9 @@ cdef class MsgpackEncoderV03(MsgpackEncoderBase):
         has_span_type = <bint> (span.span_type is not None)
         has_meta = <bint> (len(span._meta) > 0 or dd_origin is not NULL)
         has_metrics = <bint> (len(span._metrics) > 0)
+        has_parent_id = <bint> (span.parent_id is not None)
```

Comment:
This checks if span.parent_id is set, if it is, we include it in the encoding, if not, we don't encode it because the value is 0 and that's the default, so it doesn't need to be encoded.